Services/RequestServer/ConnectionFromClient.cpp

```diff
@@ -422,9 +423,10 @@ void ConnectionFromClient::start_request(i32 request_id, ByteString const& metho
             if (did_set_body && !request_headers.contains("Content-Type"))
                 curl_headers = curl_slist_append(curl_headers, "Content-Type:");
 
-            for (auto const& header : request_headers.headers()) {
-                auto header_string = ByteString::formatted("{}: {}", header.name, header.value);
+            for (auto const& [name, value] : request_headers.headers()) {
+                auto header_string = ByteString::formatted("{}: {}", name, value);
                 curl_headers = curl_slist_append(curl_headers, header_string.characters());
+                request->formatted_request_headers.append(move(header_string));
```

Comment:
According to https://curl.se/libcurl/c/curl_slist_append.html
> [curl_slist_append](https://curl.se/libcurl/c/curl_slist_append.html) copies the string.

So I don't understand what this is fixing? Did you see an ASAN error related to this?

Reply:
Ah, you're correct. Removed. 
Someone made an observation on discord in #network but I didn't verify what `curl_slist_append` was actually doing, just assumed.